api/server.go

```diff
@@ -0,0 +1,173 @@
+	err := json.NewDecoder(r.Body).Decode(&config)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	u, err := uuid.NewV4()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	newSimpleController := func(id int, status chan<- control.UserStatus) control.UserController {
+		ueConfig := userentity.Config{
+			ServerURL:    config.ConnectionConfiguration.ServerURL,
+			WebSocketURL: config.ConnectionConfiguration.WebSocketURL,
+		}
+		ue := userentity.New(memstore.New(), ueConfig)
+		return simplecontroller.New(id, ue, status)
+	}
+
+	lt := loadtest.New(&config, newSimpleController)
+	a.loadTests[u.String()] = lt
+
+	writeJsonResponse(w, map[string]string{"loadTestId": u.String()})
+}
+
+func (a *API) getLoadTestById(w http.ResponseWriter, r *http.Request) (*loadtest.LoadTester, error) {
+	vars := mux.Vars(r)
+	id := vars["id"]
+	lt, ok := a.loadTests[id]
+	if !ok {
+		err := fmt.Errorf("Load test with id %s not found", id)
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return nil, err
+	}
+	return lt, nil
+}
+
+func (a *API) runLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	if err = lt.Run(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeJsonResponse(w, map[string]string{"message": "load test started"})
+}
+
+func (a *API) stopLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	if err = lt.Stop(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeJsonResponse(w, map[string]string{"message": "load test stopped"})
+}
+
+func (a *API) destroyLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+
+	_ = lt.Stop() // we are ignoring the error here in case the load test was previously stopped
+
+	delete(a.loadTests, mux.Vars(r)["id"])
+	writeJsonResponse(w, map[string]string{"message": "load test destroyed"})
+}
+
+func (a *API) getLoadTestStatusHandler(w http.ResponseWriter, r *http.Request) {
+	_, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]string{"status": "Not yet implemented"})
+}
+
+func getAmount(r *http.Request) int {
+	amountStr := r.FormValue("amount")
+	amount := 1
+
+	if amountStr != "" {
+		if a, err := strconv.ParseInt(amountStr, 10, 16); err == nil && a > 0 {
+			amount = int(a)
+		}
+	}
+	return amount
+}
+
+func (a *API) addUserHandler(w http.ResponseWriter, r *http.Request) {
+	amount := getAmount(r)
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+
+	for i := 0; i < amount; i++ {
+		if err = lt.AddUser(); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
I am bit concerned about the case where we successfully add some users and then we get an error (reaching the max users limit for example). We return the error but we actually added some users. I am wondering if it makes more sense for most of this API to always return the `LoadTester` status and deal with the error at the JSON level by including an `err` object if something went wrong. Thoughts?